### PR TITLE
 Update documentation & Add isolated client class

### PR DESCRIPTION
**Pull request by SanduDS**
Fixes https://github.com/wso2-enterprise/choreo/issues/6660
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: twilio/Module.md [hunk 1]
```diff
@@ -1,337 +1,92 @@
...
+public function main() returns error? {
+    twilio:Account response = check twilioClient->getAccountDetails();
+}
+```
+2. Use `bal run` command to compile and run the Ballerina program.
 
-    //Send SMS remote function is called by the twilio client
-    var details = twilioClient->sendSms(fromMobile, toMobile, message);
+## Quick reference
+Code snippets of some frequently used functions:
```
Comment (RukshiW) [inline twilio/Module.md]: Suggestion: "The following are code snippets of some frequently used functions:"

File: twilio/Module.md [hunk 2]
```diff
@@ -1,337 +1,92 @@
...
+Code snippets of some frequently used functions: 
 
-    //Response is printed as log messages
-    if (details is twilio:SmsResponse) {
-        log:printInfo("SMS_SID: " + details.sid.toString() + ", Body: " + details.body.toString());
-    } else {
-        log:printInfo(details.message());
-    }
-}
+* Send an SMS
```
Comment (RukshiW) [inline twilio/Module.md]: Align the text correctly under each bullet point. This applies to all the bullet code snippets in this section.

File: twilio/Module.md [hunk 3]
```diff
@@ -1,337 +1,92 @@
...
+This section shows how to use the connector to send an SMS. You will need a verfied phone number if you are using a trial account to send the message from your twilio phone number. if the SMS is sent successfully it will provides SMSResponse record with details of the SMS otherwise it will provide the error occurred.
```
Comment (RukshiW) [inline twilio/Module.md]: Suggestion: The following snippet shows how to send an SMS via the connector. If you are using a trial account, you need a verified phone number to send the message from your Twilio phone number. If the SMS is successfully sent, it. returns the `SMSResponse` record with details of the SMS. If the SMS was not sent, it returns the error that occurred."

File: twilio/Package.md [hunk 4]
```diff
@@ -1,337 +1,19 @@
...
-```
-4. Run `./gradlew build` to build the java wrapper.
-
-5. Run `bal build twilio` to build with tests or else Run `bal build --skip-tests twilio`.
+Connects to Twilio from Ballerina
+
+## Package Overview
+The `ballerinax/twilio` is a [Ballerina](https://ballerina.io/) connector for Twilio.
+This package provides the capability to access [Twilio communication services](https://www.twilio.com/docs/all) such as
+sending an SMS, getting message detail, making a voice call etc.
```
Comment (RukshiW) [inline twilio/Package.md:6]: Change "getting message detail" to "getting message details"

File: twilio/Package.md [hunk 5]
```diff
@@ -1,337 +1,19 @@
...
+### Compatibility
+|                               | Version                              |
+|-------------------------------|--------------------------------------|
+| Ballerina Language            | Swan Lake Beta 2                     |
+| Twilio Basic API              | 2010-04-01                           |     
+
+## Report issues
+To report bugs, request new features, start new discussions, view project boards, etc., go to the [Ballerina connector repository](https://github.com/ballerina-platform/module-ballerinax-twilio)
+## Useful links
+- Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
Comment (RukshiW) [inline twilio/Package.md:17]: Change "in" to "via".

File: twilio/modules/listener/Module.md [hunk 6]
```diff
@@ -149,140 +41,61 @@ Callback URL registration method depends on the event type.
...
-        authToken: twilioAuthToken
-    };
-    twilio:Client twilioClient = new (twilioConfig);
-    var details = twilioClient->sendSms(fromNumber, toNumber, test_message, callbackUrl);
-    if (details is error) {
-        log:printInfo(details.message());
+## Quick reference
+Code snippets of some frequently used functions: 
+* QUEUED/SENT/DELIVERED SMS Event
+This examples shows how you can start a ballerina twilio listener using localhost. you will need to use ngork to expose a web server running on your local machine to the internet.
```
Comment (RukshiW) [inline twilio/modules/listener/Module.md:70]: Suggestion: "The following example shows you how to start a Ballerina Twilio listener via localhost. You need to use ngork to expose a web server running on your local machine to the internet."

File: twilio/modules/listener/Module.md [hunk 7]
```diff
@@ -149,140 +41,61 @@ Callback URL registration method depends on the event type.
...
-        accountSId: twilioAccountSid,
-        authToken: twilioAuthToken
-    };
-    twilio:Client twilioClient = new (twilioConfig);
-    var details = twilioClient->sendSms(fromNumber, toNumber, test_message, callbackUrl);
-    if (details is error) {
-        log:printInfo(details.message());
+## Quick reference
+Code snippets of some frequently used functions: 
+* QUEUED/SENT/DELIVERED SMS Event
```
Comment (RukshiW) [inline twilio/modules/listener/Module.md:69]: Have the bullet point header in title case "QUEUED/SENT/DELIVERED SMS event"

Is there a reason for the statuses to be in all caps? You have used title caps in the next bullet point header.

File: twilio/modules/listener/Module.md [hunk 8]
```diff
@@ -149,140 +41,61 @@ Callback URL registration method depends on the event type.
...
+            log:printInfo("In Progress", event);
+        }
+        remote function onCallCompleted(twilioListener:CallStatusChangeEvent event) returns error? {
+            log:printInfo("Completed", event);
+        }
     }
 }
 ```
-### Ringing/InProgress/Completed Voice Call Events
+* Ringing/InProgress/Completed Voice Call Events
```
Comment (RukshiW) [inline twilio/modules/listener/Module.md:85]: Have the bullet point header in sentence case as shown below.

"Ringing/InProgress/Completed voice call events

File: twilio/modules/listener/Module.md [hunk 9]
```diff
@@ -149,140 +41,61 @@ Callback URL registration method depends on the event type.
...
 This provides the listener support to the voice calls. you will need to use ngork to expose a web server running on your local machine to the internet. Find more sample from here.
```
Comment (RukshiW) [inline twilio/modules/listener/Module.md:86]: Suggestion: "This provides the listener support to the voice calls. You need to use ngork to expose a web server running on your local machine to the internet."

Comment (RukshiW) [reply twilio/modules/listener/Module.md:86]: Is "Find more sample from here." a duplication of the link added after all the code snippets?

File: twilio/Module.md [hunk 10]
```diff
@@ -1,337 +1,92 @@
...
-    //Twilio client
-    twilio:Client twilioClient = new (twilioConfig);
+### Step 3 - Invoke  connector operation
+1. Invoking connector operation using the client
+```ballerina
+public function main() returns error? {
+    twilio:Account response = check twilioClient->getAccountDetails();
+}
+```
+2. Use `bal run` command to compile and run the Ballerina program.
```
[the remark this answers is not in the document]
Comment (SanduDS) [reply twilio/Module.md:53]: This is in the agreed format of all connectors. I'll check this with the team

**Review by abeykoon — APPROVED** [review]

File: twilio/endpoint.bal [hunk 11]
```diff
@@ -18,15 +18,23 @@ import ballerina/auth;
...
 # + basicClient - HTTP client endpoint for basic api
 @display {label: "Twilio Client", iconPath: "logo.svg"}
-public client class Client {
-    public string accountSId;
-    public http:Client basicClient;
+public isolated client class Client {
+    private final string accountSId;
+    private final http:Client basicClient;
 
+    # Gets invoked to initialize the `connector`.
```
Comment (RukshiW) [inline twilio/endpoint.bal:31]: What gets invoked to initialize the connector? Can we have this information in the following format?

"To initialize the connector, invoke the XXX and set the API credentials."

File: twilio/endpoint.bal [hunk 12]
```diff
@@ -18,15 +18,23 @@ import ballerina/auth;
...
+    # The connector initialization requires setting the API credentials. 
+    # Create a [Twilio](https://www.twilio.com/) account and obtain account SID and auth token at [console](twilio.com/console)
```
Comment (RukshiW) [inline twilio/endpoint.bal:33]: Suggestion: "Create a [Twilio](https://www.twilio.com/) account and obtain account SID and auth token at [console](twilio.com/console). For more details, see [Twillio Documentation - Authenticate with HTTP](https://www.twilio.com/docs/usage/api#authenticate-with-http)."

File: twilio/endpoint.bal [hunk 13]
```diff
@@ -18,15 +18,23 @@ import ballerina/auth;
...
+    private final string accountSId;
+    private final http:Client basicClient;
 
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. 
+    # Create a [Twilio](https://www.twilio.com/) account and obtain account SID and auth token at [console](twilio.com/console) 
+    # or else follow this [guide](https://www.twilio.com/docs/usage/api#authenticate-with-http) for more details.
+    #
+    # + twilioConfig - Twilio connection configuration 
+    # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
```
Comment (RukshiW) [inline twilio/endpoint.bal:37]: Replace "in case of" with "in the event of a", and add a comma after "initialize".

File: twilio/endpoint.bal [hunk 14]
```diff
@@ -46,25 +54,24 @@ public client class Client {
         }
     }
 
-    # Return account details of the given account-sid.
+    # Gets account details of the given account-sid.
     #
-    # + return - If success, returns account object with basic details, else returns error
+    # + return - An account object with basic details or else an error
```
Comment (RukshiW) [inline twilio/endpoint.bal:59]: Add a comma after "details".

File: twilio/endpoint.bal [hunk 15]
```diff
@@ -46,25 +54,24 @@ public client class Client {
...
 
-    # Send SMS from the given account-sid.
+    # Sends SMS from the given account-sid.
     #
     # + fromNo - Mobile number which the SMS should be sent from
     # + toNo - Mobile number which the SMS should be received to
     # + message - Message body of the SMS
     # + statusCallbackUrl - (optional) Callback URL where the status callback events are needed to be dispatched
-    # + return - If success, returns a programmable SMS response object, else returns error
+    # + return - A programmable SMS response object or else an error
```
Comment (RukshiW) [inline twilio/endpoint.bal:74]: Add a comma after "object".

File: twilio/endpoint.bal [hunk 16]
```diff
@@ -88,10 +95,10 @@ public client class Client {
         return mapJsonToSmsResponse(payloadMap);
     }
 
-    # Get the relavant message from a given message-sid.
+    # Gets the relevant message from a given message-sid.
     #
     # + messageSid - Message-sid of a relevant message
-    # + return - If success, returns a message resource response record, else returns error
+    # + return -  A message resource response record or else an error
```
Comment (RukshiW) [inline twilio/endpoint.bal:101]: Add a comma after "record".

File: twilio/endpoint.bal [hunk 17]
```diff
@@ -103,12 +110,12 @@ public client class Client {
...
     }
 
-    # Send WhatsApp message from the given Sender ID of the account.
+    # Sends WhatsApp message from the given Sender ID of the account.
     #
     # + fromNo - Mobile number from which the WhatsApp message should be sent
     # + toNo - Mobile number by which the WhatsApp message should be received
     # + message - Message body of the WhatsApp message
-    # + return - If success, returns a WhatsAppResponse object, else returns error
+    # + return - A whatsAppResponse object or else an error
```
Comment (RukshiW) [inline twilio/endpoint.bal:118]: Add a comma after "object".

File: twilio/endpoint.bal [hunk 18]
```diff
@@ -128,15 +135,15 @@ public client class Client {
...
+    # Makes a voice call from the given account-sid.
     #
     # + fromNo - Mobile number which the voice call should be sent from
     # + toNo - Mobile number which the voice call should be received to
     # + voiceCallInput - What should be heard when the other party picks up the phone (a Url that returns TwiML Voice instructions or 
     #                    inline message. example: "http://demo.twilio.com/docs/voice.xml" or "Thank you for calling")
     # + statusCallback - (optional) StatusCallback record which contains the callback url and the events whose status 
     #                     needs to be delivered
-    # + return - If success, returns voice call response object with basic details, else returns error
+    # + return - A voiceCallresponse object with basic details or else an error
```
Comment (RukshiW) [inline twilio/endpoint.bal:146]: Add a comma after "details".

**Pull request merged**